packages/internal/src/generate/typeDefinitions.ts

```diff
@@ -32,16 +32,20 @@ import { writeTemplate } from './templates'
  * and return the generated files.
  */
 export const generateTypeDefs = async () => {
+  // MARK What does p mean?!
   const p1 = generateMirrorDirectoryNamedModules()
   const p2 = generateMirrorCells()
   const p3 = generateTypeDefRouterPages()
   const p4 = generateTypeDefCurrentUser()
   const p5 = generateTypeDefRouterRoutes()
   const p6 = generateTypeDefGlobImports()
   const p7 = generateTypeDefGlobalContext()
-  const p8 = await generateTypeDefGraphQL()
+  const p8 = generateTypeDefScenarios()
+  const p9 = await generateTypeDefGraphQL()
 
-  return [...p1, ...p2, p3[0], p4[0], p5[0], p6[0], p7[0], ...p8]
+  // MARK What is this?
+  // Why create another array
```

Comment:
I'm sorry for frustrating you! Next time please reach out to me for clarification, so that I'm aware that this code is frustrating, and so that I can adjust the code to make more sense!

This function is used by `yarn rw generate types` and `rw-gen`. It returns an array of paths to the files that it generates. 

The commands says something along the lines of:

```terminal
Generating...

 - ./redwood/types/xxx1.d.ts
 - ./redwood/types/xxx2.d.ts
 - ./redwood/types/xxx3.d.ts

```

And in the watch mode it uses the outputs: `Generating... 16 files created.`

Reply:
Whoops, meant to remove these comments. The comments _were_ there so that I could discuss!